server/util/uuid/uuid.go

```diff
@@ -43,3 +54,56 @@ func StringToBytes(text string) ([]byte, error) {
 	}
 	return uuidBytes, nil
 }
+
+func handleHostIDError(err error) {
+	hostIDError = err
+	fillHostId()
+}
+
+func fillHostId() {
+	id, err := guuid.NewRandom()
+	hostID = id.String()
+	if err != nil {
+		hostIDError = err
+		hostID = string(guuid.NodeID())
+	}
+}
+
+func GetHostID() (string, error) {
```

Comment:
The fallbacks add a lot of complexity, I think we should try to simplify this.

Can we just return errs to the caller if we encounter them?

Reply:
Yup, done.